packages/qwik/src/core/use/use-context.ts

```diff
@@ -191,29 +190,34 @@ export const createContextId = <STATE = unknown>(name: string): ContextId<STATE>
  * @public
  */
 // </docs>
-export const useContextProvider = <STATE extends object>(
-  context: ContextId<STATE>,
-  newValue: STATE
-) => {
+export const useContextProvider = <STATE>(context: ContextId<STATE>, newValue: STATE) => {
   const { val, set, elCtx } = useSequentialScope<boolean>();
+
   if (val !== undefined) {
     return;
   }
+
   if (qDev) {
     validateContext(context);
   }
+
   const contexts = (elCtx.$contexts$ ||= new Map());
+
   if (qDev && qSerialize) {
     verifySerializable(newValue);
   }
+
   contexts.set(context.id, newValue);
   set(true);
 };
 
+// default value callback function type
+type DefaultStateFn<STATE> = (value: STATE | undefined) => STATE;
+
 export interface UseContext {
-  <STATE extends object, T>(context: ContextId<STATE>, transformer: (value: STATE) => T): T;
-  <STATE extends object, T>(context: ContextId<STATE>, defaultValue: T): STATE | T;
-  <STATE extends object>(context: ContextId<STATE>): STATE;
+  <STATE, T>(context: ContextId<STATE>, transformer: (value: STATE) => T): T;
+  <STATE, T>(context: ContextId<STATE>, defaultValue: T): STATE | T;
+  <STATE>(context: ContextId<STATE>): STATE;
```

Comment:
You should add a test for number, string, etc context to make sure it works and keeps working.